crates/ruff_linter/src/rules/ruff/rules/parenthesize_logical_operators.rs

```diff
@@ -75,21 +86,47 @@ pub(crate) fn parenthesize_chained_logical_operators(
                     ..
                 },
             ) => {
+                let locator = checker.locator();
+                let source_range = bool_op.range();
                 if parenthesized_range(
                     bool_op.into(),
                     expr.into(),
                     checker.indexer().comment_ranges(),
-                    checker.locator().contents(),
+                    locator.contents(),
                 )
                 .is_none()
                 {
-                    checker.diagnostics.push(Diagnostic::new(
-                        ParenthesizeChainedOperators,
-                        bool_op.range(),
-                    ));
+                    let mut diagnostic =
+                        Diagnostic::new(ParenthesizeChainedOperators, source_range);
+                    if is_single_line_expr_with_narrow_width(
+                        locator,
+                        source_range,
+                        checker.settings,
+                    ) {
+                        let new_source = format!("({})", locator.slice(source_range));
+                        diagnostic.set_fix(Fix::applicable_edit(
+                            Edit::range_replacement(new_source, source_range),
+                            Applicability::Safe,
+                        ));
+                    }
+                    checker.diagnostics.push(diagnostic);
                 }
             }
             _ => continue,
         };
     }
 }
+
+fn is_single_line_expr_with_narrow_width(
+    locator: &Locator,
+    source_range: TextRange,
+    settings: &LinterSettings,
+) -> bool {
+    if locator.contains_line_break(source_range) {
+        return false;
+    }
+    let width_with_fix = LineWidthBuilder::new(settings.tab_size)
+        .add_str(locator.full_line(source_range.start()).trim_end())
+        .add_str("()");
+    width_with_fix <= settings.line_length
```

Comment:
I'd suggest omitting this -- we tend not to consider this except for rules that could greatly expand the width, and so it's more consistent for us to leave it as-is.

Reply:
Sure. Would we consider it an unsafe fix if it went over the line length, since it could end up triggering other rules? Or do we just not think about it at all here?

Reply:
I think "just not think about it at all here" since it's really a more general problem. I'll link some related issues in a sec...

Reply:
https://github.com/astral-sh/ruff/issues/8106

Reply:
Okiedokie, thanks!

Reply:
(You're definitely right to _care_ about this. It's just that it's something we need to solve globally, and I want to avoid adding special-casing in each rule for now.)

Reply:
Sure, makes sense!